official/vision/modeling/backbones/resnet.py

```diff
@@ -293,12 +297,18 @@ def __init__(
         block_fn = nn_blocks.BottleneckBlock
       else:
         raise ValueError('Block fn `{}` is not supported.'.format(spec[0]))
+      use_first_projection = (
+        spec[0] == 'bottleneck'
+        or i > 0
+        or (i == 0 and self._use_first_projection)
```

Comment:
Thanks for the update. The logic is much more clearer now.  Is the `i==0` necessary for the last line?
Is it equivalent to to just do
`
use_first_projection = (
        spec[0] == 'bottleneck'
        or i > 0
        or self._use_first_projection)
`

Reply:
yes I think you are right! corrected